src/ResourceList.js

```diff
@@ -100,266 +29,152 @@ H5P.ResourceList = (function () {
+    this.wrapper.classList.add('h5p-resource-list-wrapper');
+
+    const buttonContent = document.createElement('div');
+    buttonContent.className = 'h5p-resource-list-button-content';
+
+    const headerIcon = document.createElement('img');
+    headerIcon.className = 'h5p-resource-list-button-image';
+    headerIcon.alt = ''; // Merely decorational
+    headerIcon.src = resourceImage;
+    buttonContent.appendChild(headerIcon);
+
+    const buttonText = document.createElement('span');
+    buttonText.className = 'h5p-resource-list-button-text';
+    buttonText.innerText = this.l10n.resources;
+    buttonContent.appendChild(buttonText);
+
+    const readIcon = document.createElement('span');
+    readIcon.className = 'fa fa-arrow-right';
+    buttonContent.appendChild(readIcon);
+
+    const button = document.createElement('button');
+    button.type = 'button';
+    button.addEventListener('click', this.toggleResources.bind(this));
+    button.className = 'h5p-resource-list-button';
+    button.appendChild(buttonContent);
+    button.setAttribute('aria-label', this.l10n.resourcesLabel);
+    this.wrapper.appendChild(button);
+
+    this.listContainer = document.createElement('div');
+    this.listContainer.classList.add('h5p-resource-list-container');
+
+    this.listContainer.appendChild(createHeader(this.l10n, this.toggleResources.bind(this)));
+    this.listContainer.appendChild(createList(this.id, this.l10n, this.params.resourceList));
+    this.listContainer.classList.add('hidden');
+
+    this.wrapper.appendChild(createBackground(this.toggleResources.bind(this)));
+    this.wrapper.appendChild(this.listContainer);
+
+    this.container.appendChild(this.wrapper);
+    setTimeout(this.resize(), 0);
+  }
 
-      wrapper.appendChild(createBackground());
-      wrapper.appendChild(listContainer);
+  /**
+   * Toggle display of resource list
+   */
+  toggleResources() {
+    const focusableElementsString = 'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex="0"], [contenteditable]';
 
-      this.container.appendChild(wrapper);
-      setTimeout(this.resize, 0);
-    };
+    const isActive = this.wrapper.classList.contains('h5p-resource-list-active');
+    if (!isActive) {
+      const focusableElements = Array.from(this.listContainer.querySelectorAll(focusableElementsString));
+      this.wrapper.onkeydown = (event) => trapKeys(event, focusableElements[0], focusableElements[focusableElements.length - 1], this.toggleResources);
+      this.listContainer.classList.remove('hidden');
+      this.listContainer.classList.remove('slide-out');
+      this.listContainer.classList.add('slide-in');
+    }
+    else {
+      this.wrapper.onkeydown = () => { };
+      this.listContainer.classList.remove('slide-in');
+      this.listContainer.classList.add('slide-out');
+      setTimeout(() => this.listContainer.classList.add('hidden'), 500);
+    }
```

Comment:
Would be nice to swap this if/else.

```suggestion
    if (!isActive) {
      this.wrapper.onkeydown = () => { };
      this.listContainer.classList.remove('slide-in');
      this.listContainer.classList.add('slide-out');
      setTimeout(() => this.listContainer.classList.add('hidden'), 500);
    }
    else {
      const focusableElements = Array.from(this.listContainer.querySelectorAll(focusableElementsString));
      this.wrapper.onkeydown = (event) => trapKeys(event, focusableElements[0], focusableElements[focusableElements.length - 1], this.toggleResources);
      this.listContainer.classList.remove('hidden');
      this.listContainer.classList.remove('slide-out');
      this.listContainer.classList.add('slide-in');
    }
```

Reply:
Yess